bin/modules/untappd/untapped_module.dart

```diff
@@ -0,0 +1,258 @@
+  void _checkUntappd() async {
+    if (!_isInitialized) {
+      print('Untappd module not initialized!');
+      throw Exception('Untappd module not initialized!');
+    }
+    var box = Hive.box(HiveConstants.untappdBox);
+
+    Map<dynamic, dynamic> listOfUsers =
+        await box.get(HiveConstants.untappdUserList, defaultValue: {});
+    var latestCheckins = await box
+        .get(HiveConstants.untappdLatestUserCheckins, defaultValue: {});
+
+    var updateChannelId = await box.get(HiveConstants.untappdUpdateChannelId);
+
+    if (updateChannelId == null) {
+      print('No channel available for updates!');
+      return;
+    }
+
+    if (listOfUsers.isEmpty) print('No users available to scrape!');
+
+    listOfUsers.forEach((userSnowflake, untappdUsername) async {
+      var latestCheckinDisk = latestCheckins[untappdUsername];
+      try {
+        var latestCheckinUntappd = await _getLatestCheckin(untappdUsername);
+
+        // If a new ID is available, post update!
+        if (latestCheckinUntappd != null &&
+            latestCheckinDisk != latestCheckinUntappd.id) {
+          // Update latest saved checkin
+          latestCheckins.addAll({untappdUsername: latestCheckinUntappd.id});
+          await box.put(
+              HiveConstants.untappdLatestUserCheckins, latestCheckins);
+
+          // Build update message with info from untappd checkin
+          var user = await _bot.fetchUser(Snowflake(userSnowflake));
+          var embedBuilder = EmbedBuilder();
+          embedBuilder.title = '${user.username} is drinking beer!';
+          embedBuilder.url =
+              _getCheckinUrl(latestCheckinUntappd.id, untappdUsername);
+          embedBuilder.description = latestCheckinUntappd.title;
+          if (latestCheckinUntappd.comment.isNotEmpty) {
+            embedBuilder.addField(
+                field:
+                    EmbedFieldBuilder('Comment', latestCheckinUntappd.comment));
+          }
+          if (latestCheckinUntappd.rating.isNotEmpty) {
+            embedBuilder.addField(
+                field: EmbedFieldBuilder(
+                    'Rating',
+                    _buildRatingEmoji(
+                        double.parse(latestCheckinUntappd.rating))));
+          }
+          if (latestCheckinUntappd.photoAddress != null) {
+            embedBuilder.imageUrl = latestCheckinUntappd.photoAddress;
+          }
+
+          // Get channel used for untappd updates, previously set by discord admin.
+          var updateChannel = await _bot
+              .fetchChannel(Snowflake(updateChannelId))
+              .then((value) => (value as ITextChannel));
+
+          // Send update message
+          await updateChannel.sendMessage(MessageBuilder.embed(embedBuilder));
+        }
+        // Sleep 5 seconds per user to avoid suspicious requests to untappd server
+        await Future.delayed(Duration(seconds: 5));
+      } catch (e) {
+        print(e.toString());
+      }
```

Comment:
**Suggestion:** Consider handling exceptions in a more user-friendly way. Instead of just printing the error, you could log it and show a user-friendly message to the user.
```suggestion
      catch (e) {
      Logger().e('Error while checking Untappd: $e');
      throw UserFriendlyException('An error occurred while checking Untappd.');
      }
```